File: app-mvvm/src/main/java/com/xuie/imaginary/data/source/NetsSource.java
package xuk.imaginary.data.source;

import xuk.imaginary.data.NetsDetail;
import xuk.imaginary.data.NetsSummary;
import xuk.imaginary.data.VideoBean;

import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Single;

/**
 * @author xuie
 * @date 17-8-17
 */
public interface NetsSource {
    Single<List<NetsSummary>> getNews(int page);

    Observable<NetsDetail> getNewDetail(String postId);

    Single<List<VideoBean>> getVideoList(String type, int page);
}

File: app-coroutines/src/main/kotlin/xuk/imaginary/data/source/NetsSource.kt
package xuk.imaginary.data.source

import xuk.imaginary.data.NetsDetail
import xuk.imaginary.data.NetsSummary
import xuk.imaginary.data.VideoBean

import io.reactivex.Observable
import io.reactivex.Single

/**
 * @author Jie Xu
 * @date 17-8-17
 */
interface NetsSource {
  fun getNews(page: Int): Single<List<NetsSummary>>

  fun getNewDetail(postId: String): Observable<NetsDetail>

  fun getVideoList(type: String, page: Int): Single<List<VideoBean>>
}
